docs/tutorials/intro_tutorial.ipynb

```diff
@@ -706,22 +717,28 @@
     "        self.model.grid.move_agent(self, new_position)\n",
     "\n",
     "    def give_money(self):\n",
-    "        if self.wealth > 0: \n",
-    "            cellmates = self.model.grid.get_cell_list_contents([self.pos])\n",
-    "            if len(cellmates) > 1:\n",
-    "                other_agent = self.random.choice(cellmates)\n",
-    "                other_agent.wealth += 1\n",
-    "                self.wealth -= 1\n",
+    "        cellmates = self.model.grid.get_cell_list_contents([self.pos])\n",
+    "        # Ensure agent is not giving money to itself\n",
+    "        cellmates.pop(cellmates.index(self))\n",
+    "        if len(cellmates) > 0:\n",
+    "            other_agent = self.random.choice(cellmates)\n",
+    "            other_agent.wealth += 1\n",
+    "            self.wealth -= 1\n",
+    "\n",
```

Comment:
Per the comment, I think this could either be left as "step" common convention with an explicit comment on options or

make it consistent where it would be 
`
def step(self):
        self.datacollector.collect(self)
        self.agents.shuffle_do("move")
        self.agents.do("give_money")
`